src/apphosting/secrets/dialogs.ts

```diff
@@ -0,0 +1,148 @@
+import * as clc from "colorette";
+const Table = require("cli-table");
+
+import * as apphosting from "../../gcp/apphosting";
+import * as prompt from "../../prompt";
+import * as iam from "../../gcp/iam";
+import * as utils from "../../utils";
+import { logger } from "../../logger";
+
+interface Metadata {
+  location: string;
+  id: string;
+  accounts: string[];
+}
+
+export function serviceAccountsForBackend(
+  projectNumber: string,
+  backend: apphosting.Backend,
+): string[] {
+  if (backend.serviceAccount) {
+    return [backend.serviceAccount];
+  }
+  return [
+    iam.getDefaultCloudBuildServiceAgent(projectNumber),
+    iam.getDefaultComputeEngineServiceAgent(projectNumber),
+  ];
+}
+
+export function toMetadata(projectNumber: string, backends: apphosting.Backend[]): Metadata[] {
+  const metadata: Metadata[] = [];
+  for (const backend of backends) {
+    const [, , , location, , id] = backend.name.split("/");
+    metadata.push({ location, id, accounts: serviceAccountsForBackend(projectNumber, backend) });
+  }
+  return metadata.sort((left, right) => {
+    const cmplocation = left.location.localeCompare(right.location);
```

Comment:
If we're sorting, consider adding a second sort column for name (will help with users with multi-regional backends)

Reply:
That's what the next condition does (`return left.id.localeCompare(right.id)`)